trunk/workflow-manager/src/main/java/org/mitre/mpf/wfm/camel/operations/detection/trackmerging/TrackMergingProcessor.java

```diff
@@ -142,26 +173,44 @@ private TrackMergingPlan createTrackMergingPlan(String samplingIntervalProperty,
 				}
 			} catch (Exception exception) {
 				log.warn("Attempted to parse " + MpfConstants.MEDIA_SAMPLING_INTERVAL_PROPERTY + " value of '{}' but encountered an exception. Defaulting to 1 and disabling track merging.", samplingIntervalProperty, exception);
-				return new TrackMergingPlan(1, false);
+				return new TrackMergingPlan(1, 1, false, 1);
+			}
+		}
+		if (minTrackLengthProperty != null) {
+			try {
+				minTrackLength = Integer.valueOf(minTrackLengthProperty);
+			} catch (Exception exception) {
+				log.warn("Attempted to parse "+ MpfConstants.MIN_TRACK_LENGTH +" value of '{}' but encountered an exception. Defaulting to 1.", mergeTracksProperty, exception);
+				return new TrackMergingPlan(1, 1, false, 1);
 			}
 		}
 		if (mergeTracksProperty != null) {
 			try {
 				mergeTracks = Boolean.valueOf(mergeTracksProperty);
 			} catch (Exception exception) {
-				log.warn("Attempted to parse "+MpfConstants.MERGE_TRACKS_PROPERTY+" value of '{}' but encountered an exception. Defaulting to false.", mergeTracksProperty, exception);
-				return new TrackMergingPlan(1, false);
+				log.warn("Attempted to parse "+ MpfConstants.MERGE_TRACKS_PROPERTY +" value of '{}' but encountered an exception. Defaulting to false.", mergeTracksProperty, exception);
+				return new TrackMergingPlan(1, 1, false, 1);
+			}
+		}
+		if (minGapBetweenTracksProperty != null) {
+			try {
+
+				minGapBetweenTracks = Integer.valueOf(minGapBetweenTracksProperty);
+			} catch (Exception exception) {
+				log.warn("Attempted to parse "+ MpfConstants.MIN_GAP_BETWEEN_TRACKS+ " value of '{}' but encountered an exception. Defaulting to 1.", mergeTracksProperty, exception);
+				return new TrackMergingPlan(1, 1, false, 1);
 			}
 		}
-		return new TrackMergingPlan(samplingInterval, mergeTracks);
+		return new TrackMergingPlan(samplingInterval, minTrackLength, mergeTracks, minGapBetweenTracks);
 	}
 
-	private Set<Track> combine(Set<Track> sourceTracks, int samplingInterval, double segMinOverlap) {
+	private Set<Track> combine(Set<Track> sourceTracks, TrackMergingPlan plan, double segMinOverlap) {
```

Comment:
There are other problems with this line that I found on investigation.

> SortedSet<Track> newTracks = new TreeSet<Track>(combine(tracks, trackMergingPlan, propertiesUtil.getTrackOverlapThreshold()));

We're pulling the segMinOverlap directly from the properties file, ignoring settings on the action or job.  I'm going to open a new task for this, so we can be sure it's tracked and tested appropriately.